fdbclient/AuditUtils.actor.cpp

```diff
@@ -29,6 +29,568 @@
 
 #include "flow/actorcompiler.h" // has to be last include
 
+// Check if any pair of ranges are exclusive with each other
+// This is not a part in consistency check of audit metadata
+// This is used for checking the validity of inputs to rangesSame()
+bool elementsAreExclusiveWithEachOther(std::vector<KeyRange> ranges) {
+	ASSERT(std::is_sorted(ranges.begin(), ranges.end(), KeyRangeRef::ArbitraryOrder()));
+	for (int i = 0; i < ranges.size() - 1; ++i) {
+		if (ranges[i].end > ranges[i + 1].begin) {
+			TraceEvent(SevError, "ElementsAreNotExclusiveWithEachOther").detail("Ranges", describe(ranges));
```

Comment:
Remove this `TraceEvent(SevError`, we don't need to generate an error in a generic validation function, the caller can generate necessary events.